controllers/runner_controller.go

```diff
@@ -872,6 +873,15 @@ func newRunnerPod(template corev1.Pod, runnerSpec v1alpha1.RunnerConfig, default
 		}...)
 	}
 
+	if len(pod.Spec.ImagePullSecrets) == 0 && len(defaultRunnerImagePullSecrets) > 0 {
+		// runner spec didn't provide custom values and default image pull secrets are provided
+		for _, imagePullSecret := range defaultRunnerImagePullSecrets {
+			pod.Spec.ImagePullSecrets = []corev1.LocalObjectReference{
+				{Name: imagePullSecret},
+			}
```

Comment:
```suggestion
			pod.Spec.ImagePullSecrets = append(pod.Spec.ImagePullSecrets, []corev1.LocalObjectReference{
				{Name: imagePullSecret},
			})
```

otherwise pod.spec.ImagePullSecrets always contain one element which correlate to the last element in defualtRUnnerImagePullSecrets, which won't be what you want :)

Reply:
Thanks good catch - fixed in ffa0d582a28a3eeac825fadd4efede5189d6141c